### ISSUE TITLE
Update *.domain.tld as well as @.domain.tld

### ISSUE THREAD

**Issue by cdlm** [kofemann/livedns4synology#3]
Gandi supports wildcard `*` subdomains, but they should be updated as well as the `@`. I hacked the script to achieve that for me, but a clean implementation should be devised…

Given that the configuration form in DSM validates the hostname a little strictly, and that there is no other setting (I wouldn't abuse the `username` field for that), maybe make two versions of the script?

**Comment by kofemann** [comment]
You can make a PR as is and we discuss it as a review.

**Comment by kofemann** [comment]
Thanks for the RP. Do you have a link to the documentation related to `*`. TO me it unclear what it actually does. and what is the difference to `@`.

**Comment by cdlm** [comment]
Here's a passage in the docs: https://docs.gandi.net/en/domain_names/faq/record_types/a_record.html#what-are-wildcard-a-records. So `@` means *no* subdomain (just the main `example.tld` you're paying for), and `*` means *any* subdomain (that doesn't have an explicit record otherwise).

Reading this, I realize that I don't rely on dynamic subdomains other than by laziness, so I'll probably/eventually set up a proper zone update script for when I add/remove a vhost…

**Comment by kofemann** [comment]
Ok, obviously '@' and '*' are different usecases. Thus the proposed change is wrong.

Is the handling onf wildcard records makes sense is a different question.

**Comment by cdlm** [comment]
> Ok, obviously '@' and '*' are different usecases.

Agreed. Here's a new version, untested because I currently don't have access to my NAS.
The idea is either you use it as previously by giving `foo.example.tld` in DSM, or just `example.tld` in which case the `@.example.tld.` A record will be updated.

Or, by writing a wrapper script that passes additional arguments at the end, you can update explicit hosts, including the special `@` and `*`:
```bash
/bin/bash
./livedns.py "$@" foo bar @ '*'
```

What do you think?
